lib/web_ui/flutter_js/src/entrypoint_loader.js

```diff
@@ -151,7 +151,13 @@ export class FlutterEntrypointLoader {
       if (this._ttPolicy != null) {
         jsSupportRuntimeUri = this._ttPolicy.createScriptURL(jsSupportRuntimeUri);
       }  
-      const dartModulePromise = WebAssembly.compileStreaming(fetch(moduleUri));
+
+      const stringBuiltinSupported = this._detectWasmStringBuiltin();
+
+      const dartModulePromise = WebAssembly.compileStreaming(
```

Comment:
We may consider in the future to make the `.mjs` file export 3 things:

* (new) Compile API wrapping `WebAssembly.compile*` 
* Instantiate API wrapping `WebAssembly.instantiate*` (we already provide this)
* Invoke `main` API (we already provide this)

That would encapsulate this logic here in the dart2wasm compiler instead of making a user of the compiler aware of what builtins the compiled code requires.

There's an aspect to think about: The current way allows flutter JS code to be loaded in parallel to the mjs file, then we'd have a dependency from former to the latter - which may harm startup latency. This may especially be noticeable if we use imported strings (which may make the `mjs` file much bigger). So we'd have to think whether it would make sense to do that or not.

Anyways, that's not needed right now.

Reply:
Why do you want separate compilation and instantiation steps? Currently we don't do anything with the compiled module other than instantiating it. We could combine both steps in a single export.

Reply:
We can also offer this in the mjs api:

* (new) Compile&Instantiate API wrapping `WebAssembly.compile*` and `WebAssembly.instantiate*`

The setup should be optimized for startup time / initial page load. So we should try to do as much in parallel we can. So one advantage of separating the two, is that it allows the browser to download the wasm file & do some work in the streaming compile step in parallel to downloading/parsing/running the mjs file (which may become bigger once we have imported strings).

I don't know if Chrome/V8 does a lot of work in the compile step or most in instantiation step. May also differ between browsers.

This would need to be measured.

Reply:
> So one advantage of separating the two, is that it allows the browser to download the wasm file & do some work in the streaming compile step in parallel to downloading/parsing/running the mjs file (which may become bigger once we have imported strings).

If the compile function is provided by the mjs file so you can't compile the Wasm and download mjs file with lots of strings in parallel, right?

Maybe strings can be in a separate file that we download in parallel.